src/server/controllers/match.controller.js

```diff
@@ -0,0 +1,127 @@
+'use strict';
+
+const matchSchema = require('mongoose').model('match');
+import mailer from 'nodemailer';
+
+/*
+ * Methods about mentoring request, accept or reject including E-mail Service
+ */
+
+let matchData = new matchSchema();
+
+let sampleSuccessResult = {
+  successCode: 1,
+  matchData: matchData,
+  requestList: {
+    accepted: [
+      {
+        mentorId: 'abc123abc123',
+        mentorName: 'Jung Su Kim',
+        mentorPic: 'http://789ghi012jkl.jpg',
+      },
+    ],
+    requested: [
+      {
+        menteeId: 'abc456abc456',
+        menteeName: 'Marry Su',
+        menteePic: 'http://123abc456def.png',
+      },
+      {
+        menteeId: 'aaa111aaa111',
+        menteeName: 'Jane Park',
+        menteePic: 'http://123abc456def.png',
+      },
+    ],
+    pending: [
+      {
+        mentorId: 'aaa222aaa222',
+        mentorName: 'Sun Lee',
+        mentorPic: 'http://789ghi012jkl.jpg',
+      },
+      {
+        mentorId: 'aaa333aaa333',
+        mentorName: 'Harry Potter',
+        mentorPic: 'http://789ghi012jkl.jpg',
+      },
+    ],
+  },
+};
+
+let sampleFailResult = {
+  successCode: 0,
+  errPoint: null,
+  err: null,
+};
+
+// Send mentoring request pushing Email to mentor(receiver)
+function sendRequestEmail(req, res, receiver) {
+  let transport
+    = mailer.createTransport('smtps://yoda.mentor.lab%40gmail.com:svbootcamp@!@smtp.gmail.com');
+  let mailOptions = {
+    from: '"Yoda Service Team" <yoda.mentor.lab@gmail.com>',
+    to: receiver,
+    subject: 'New mentee needs your help!',
+    text: 'Hi, mentor!',
+    html: '<h1>Hi, ' + matchData.mentorName
+    + ', </br>new mentee needs your mentoring.</h1><p> - ' + matchData.menteeName + ' sented</p>',
+  };
+
+  transport.sendMail(mailOptions, function(err, response){
+    if (err){
+      sampleFailResult.errPoint = 'RequestMentoring - transport.sendMail';
+      sampleFailResult.err = err;
+
+      res.json(sampleFailResult);
+    } else {
+      sampleSuccessResult.mailOptions = mailOptions;
+
+      res.json(sampleSuccessResult);
+    }
+    transport.close();
+  });
+}
+
+// The mentee sent request to Mentor
+export function requestMentoring(req, res, next) {
+  matchData.mentorId = req.body.mentorId; // Email
```

Comment:
Each mentor/mentee also has user_id we defined in the user model. Can we use that? 

@SarahhhC : This is actually related to the get_mentor() method you're building, where you're extracting a subset of user object (id, email, name) and only pass that around instead of the entire user object. We have 2 options:

1) Pass around the entire user object: this makes the rest of the backend code much easier because we have the full record of the users. So we can just do the matching by 2 users (mentor/mentee). The disadvantage is that our system can get a bit bloated since you'll be returning back all the user data from db. 

2) Define the list of fields that are useful for the backend / frontend. At some point, we may decide to store the user preferences (fact that s/he likes cooking, prefers looking at mentors from good schools etc) which may not be used for the UI or any other matching service. In that case, we can decide to return back a subset of fields that we absolutely need (for example, for the "detailed view" page). If we need more fields for small set of users, we can make another db lookup to get the full set of fields. The upside of this is that the system is light. The downside is, whatever that's not included in the "user preview" object, will have to be retrieved again using find(user_id).

@KaylaKwon : Either way works fine, but the matchData should then have to store 2 objects (mentor, mentee) which is either the full user object, or a subset of user object. Let's avoid having another custom data fields like menteeId, mentorName, etc.